### DISCUSSION TITLE
SSG - First timer

### DISCUSSION THREAD

**Discussion by stephenmeehanuk** [statamic/ssg#112] [Q&A]
Hi, 

I'm trying to use SSG to create a micro website that'll be used at a location where there is no internet connection, so this particular site has to work 100% local.

My hope is I'll be able to run the website by simply dragging the index.html file into the browser and not need to run a local web server.

SSG is working, but I've got a couple of issues.

----

SSG outputs subpages in folders like this,

```
index.html
delivering-infrastructure-resilience/index.html
```

This is a problem when running the site locally, as I can't seem to link to these pages... The URLs generated by SSG look like this

```
href="/delivering-infrastructure-resilience"
```

Is there a way to output all the pages into the root folder insead? So it'd look more like this?

```
index.html
delivering-infrastructure-resilience.html
```
and make sure the `href` looks like this?

```
href="delivering-infrastructure-resilience.html"
```

I've added these to `ssg.php`
```
  'urls' => [     
        '/delivering-safer-quieter-and-easier-journeys/',
        '/informing-every-journey/',
        '/delivering-infrastructure-resilience/',
        '/delivering-the-digital-railway/',
        '/maximising-value-for-operators/'
    ],
  ```

But they don't seem to have any affect when viewing the site locally.

Is there a way to get these links to work?

  
----

The other issue I have is linking to CSS and JS

`index.html` contains this 

```
<link rel="stylesheet" href="/css/tailwind.css">
```
When I drag index.html into the browser, the css file is missing?

It'll only work I remove the first forward slash
  ```
<link rel="stylesheet" href="css/tailwind.css">
```

The other pages inside subfolders have a similar issue, but require a different fix

```
delivering-infrastructure-resilience/index.html
```
Needs the link tag to look in the root

```
<link rel="stylesheet" href="../css/tailwind.css">
```
I need to apply this to every image and video link too, which isn't ideal.

---

Is there a way around this?

**Comment by duncanmcclean** [comment]
Regarding the first URL issue... what if you simply change the URL structure in your collection settings to include the `.html` at the end, does that work (it's possible it won't)?

I'd expect that behaviour regarding the CSS & JS as if it has a `/` at the start of the path the browser will assume you want the resource from the root of the domain, instead of relative to the current folder you're in. Not sure the best way of fixing that for assets 🤔

**Reply by stephenmeehanuk** [reply]
Thanks for the suggestion, you mean like this?
```
{parent_uri}/{slug}.html
```
Unfortunately, it doesn't work. I get a 404 error.

**Reply by stephenmeehanuk** [reply]
I guess, I'm looking for a way to output all pages in the root folder, instead of sub folders. And paths to assets to not include `/` at the start. Could this functionality be added to SSG? Totally appreciate this is super niche!

**Comment by jasonvarga** [comment]
> My hope is I'll be able to run the website by simply dragging the index.html file into the browser and not need to run a local web server.

I don't think it will ever support this, sorry.

You'll need at least a super basic web server. You can just built in PHP by doing `php -S localhost:3000`
